rust/src/arrow/kernels.rs

```diff
@@ -27,40 +27,67 @@ use arrow_schema::DataType;
 
 use crate::{Error, Result};
 
+pub trait NumericMaxMin {
+    fn max_value() -> Self;
+    fn min_value() -> Self;
+}
+
+macro_rules! impl_numeric_max_min {
+    ($ty:ty) => {
+        impl NumericMaxMin for $ty {
+            fn max_value() -> Self {
+                Self::MAX
+            }
+            fn min_value() -> Self {
+                Self::MIN
+            }
+        }
+    };
+}
+
+impl_numeric_max_min!(f32);
+impl_numeric_max_min!(f64);
+impl_numeric_max_min!(i16);
+impl_numeric_max_min!(u32);
```

Comment:
Looks good, lets just fill them with the rest of the numeric types?

Maybe we should also add support for two 16bit floats too.

Reply:
Just realized this exact trait already exists in num_traits as `Bounded`, so replaced with that.